website/content/docs/nia/terraform-modules.mdx

```diff
@@ -0,0 +1,123 @@
+
+```hcl
+variable "services" {
+  description = "Consul services monitored by Consul-Terraform-Sync"
+  type = map(
+    object({
+      id        = string
+      name      = string
+      kind      = string
+      address   = string
+      port      = number
+      meta      = map(string)
+      tags      = list(string)
+      namespace = string
+      status    = string
+
+      node                  = string
+      node_id               = string
+      node_address          = string
+      node_datacenter       = string
+      node_tagged_addresses = map(string)
+      node_meta             = map(string)
+
+      cts_user_defined_meta = map(string)
+    })
+  )
+}
+```
+
+Keys of the `services` map are unique identifiers of the service across Consul agents and data centers. Keys follow the format `service-id.node.datacenter` (or `service-id.node.namespace.datacenter` for Consul Enterprise). A complete list of attributes available for the `services` variable is included in the [documentation for Consul-Terraform-Sync tasks](/docs/nia/tasks#task-execution).
+
+## How to Create a Compatible Terraform Module
+
+You can read more on how to [create a module](https://www.terraform.io/docs/modules/index.html) or work through a [tutorial to build a module](https://learn.hashicorp.com/tutorials/terraform/module-create). Consul-Terraform-Sync is designed to integrate with any module that satisfies the specifications in the following section.
+
+The repository [hashicorp/consul-terraform-sync-template-module](https://github.com/hashicorp/consul-terraform-sync-template-module) can be cloned and used as a starting point for structuring a compatible Terraform module.
+
+First, create a directory to organize Terraform configuration files that make up the module. You can start of with creating two files `main.tf` and `variables.tf` and expand from there based on your module and network infrastructure automation needs. The `main.tf` is the entry point of the module and this is where you can begin authoring your module. It can contain multiple Terraform resources related to an automation task that uses Consul service discovery information, particularly the required `services` input variable.
+
+The code example below shows a resource using the `services` variable. When this example is used in automation with Consul-Terraform-Sync, the content of the local file would dynamically update as Consul service discovery information changes.
+
+```hcl
+# Create a file with service names and their node addresses
+resource "local_file" "consul_services" {
+  content = join("\n", [
+    for _, service in var.services : "${service.name} ${service.id} ${service.node_address}"
+  ])
+  filename = "consul_services.txt"
+}
+```
+
+For the second file, copy the `services` variable declaration, found above, to the `variables.tf` file. Your module can optionally have other [variable declarations](#module-input-variables) in addition to `var.services`.
```

Comment:
> For the second file, copy the `services` variable declaration, found above, to the `variables.tf` file. 

From how I'm understanding this, I think the two files might be referring to the same file. If so, maybe something like "For the `variables.tf` file, copy the...". I was also unsure about the "second file" since a "first file" wasn't mentioned earlier.

It also looks like `services` variable declaration is 'below' instead of 'above'?

Reply:
This was difficult for me to word. I was trying to avoid copying the var.services above within the Module Specification section. However this confusion lead me to think it'll be better to reorganize overall instead of rewording it.

I moved the var.services into a new section to group the ideas better.